### DISCUSSION TITLE
Enable single file upload having size larger than 100mb at once

### DISCUSSION THREAD

**Discussion by Nikiita2k** [community/community#68] [Other Feature Feedback, Questions, & Ideas] [answered]
Currently Github doesn't supports uploading large files at once. I've tried both browser and Github desktop app, but in terms of browser, it's restricted to 25mb & for Github app, it's 100mb. It would be great if Github adds the support to upload 100-500 mb files at once!

**Comment by Neilblaze** [comment, marked as answer]
Github does support uploading and managing large files at once. Check out Git LFS. Using this you can manage/upload & control your project/repo having sizes larger than 100mb, noting that tha maximum repository size is less than equal to 10GB for free tier. If you want increase the size further, Github provides organizational level grants where it's a pay as you scale model.

Git LFS -> https://git-lfs.github.com/
Also you can check Git Kraken -> https://www.gitkraken.com/ (In case if you face issues managing your git)

**Reply by Nikiita2k** [reply]
Thanks @Neilblaze

**Reply by SuperStar518** [reply]
@Neilblaze You saved my day!

**Reply by DryreL** [reply]
Git LFS is not efficient. GitHub definetely needs to support 100-500 MB at once.
As a game designer, 100 MB hard limit on a single file is not enough for Unreal Engine 5 game development. For example, MetaHuman's head mesh is only 150 MB. Or the landscapes can easily become more than 100 MB.

**Reply by ssproessig-thales** [reply]
Paying customers should not be limited in the file size, at least not to 100MB. Sometimes you can't choose on the structure of the repo and just have to live with it - and LFS may not be available in some upstream repo.

**Reply by AfaqGhauri** [reply]
how to upload that file greater than 100mb for game developers?

**Reply by marekmrva-ss** [reply]
This actually prevents us from migrating repos from other services that actually support 200MB semi-large files to GitHub.

For game development specifically, we have 3rd party libraries that are in the 100-200MB range. Migrating to LFS would mean changing history, which is impractical as soon as commit signing is concerned.

It would be _really awesome_ if the limit could be bumped, at least for paying customers, to 200MB. That would instantly solve all our problems

**Comment by Rohinivv96** [comment]
I tried to upload weight file of object detection on Git-hub using git LFS and also using Git-hub desktop.
But still it is showing that "You can not upload file greater than 100mb".
What can be the other solution ?
Thank you.

**Reply by CodeAura** [reply]
i have the same problem

**Reply by MradulGames** [reply]
you can put .weight in back of the file that shows the size error and after that its commit but there is a problem if you put .weight then some texture are not visible I don't know answer of that but yes if you put .weight after the file shows error will commit

**Reply by Atharva-debug** [reply]
have you found a solution?

**Comment by CodeAura** [comment]
so basicly i had a big file added to my repo, but i deleted it. I tried to push my commit but it didnt work?

How can i fix this?

![image](https://user-images.githubusercontent.com/89779006/175249189-f6b12e2b-ab8c-46fd-b787-dd5a735e19d4.png)

**Reply by fredw60** [reply]
any updates here?

**Reply by DryreL** [reply]
Remove your last commit

git reset HEAD~n (n = last n commits - change here - for instance type 1)
git push origin -f

**Reply by airtower-luna** [reply]
That should've been a new thread, but in short: Removing the file with a new commit won't remove it from history. You need to edit the history so the large file is gone from all commits. There are [semi-automated ways to do it](https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-large-files-on-github#removing-files-from-a-repositorys-history), if you want to learn how to do it the hard (manual) way check [Git Tools – Rewriting History](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History).

**Comment by GaurDev06** [comment]
* After having downloaded and installed git lfs in your system, type in the command on your command line: git lfs install
* Then, type git lfs track “<extension of the file which you want to add>”. This step ensures that the git lfs supports the kind of file that you want to push to your repository.
Example: git lfs track “*.weights”
* Then type in git add .gitattributes
* Then, you have to add the file which you want to push.
Example: git add yolov3.weights
* Then, you commit the file.
Example: git commit -m “First commit”
You can add any commit message in this step inside the “”.
* Copy the URL of your remote repository.
Example : https://github.com/your-username/your-remote-repository.git
* Type in the command on your command line (or, git bash):
git remote add origin <the url>
* Finally, push the code to your repository:
git push -u origin master
Hope it helps.

**Reply by Welding-Torch** [reply]
This answer saved me a problem I've been stuck on FOR HOURS. Thank you so much.
